pkg/didcomm/protocol/didexchange/persistence.go

```diff
@@ -175,25 +183,34 @@ func (c *ConnectionRecorder) GetConnectionRecordByNSThreadID(nsThreadID string)
 // saveConnectionRecord saves the connection record against the connection id  in the store
 // TODO - https://github.com/hyperledger/aries-framework-go/issues/622 Transient v Permanent Data store
 func (c *ConnectionRecorder) saveConnectionRecord(record *ConnectionRecord) error {
-	if err := c.marshalAndSave(connectionKeyPrefix(record.ConnectionID), record); err != nil {
+	if err := marshalAndSave(connectionKeyPrefix(record.ConnectionID), record, c.transientStore); err != nil {
 		return fmt.Errorf("save connection record: %w", err)
 	}
 
 	if record.State != "" {
```

Comment:
Unrelated to this PR, but when does record.State != "" happen?

Perhaps we could use a constant rather than "" that describes this situation?

Reply:
@sudeshrshetty @rolsonquadras

Reply:
@troyronda had a look at the code. This is never happens - when we create the record first time we are adding a state(caller of this function). Probably I had added as a safety net.